### DISCUSSION TITLE
subfolder approach won't work for me

### DISCUSSION THREAD

**Discussion by dbill23** [lorinkoz/django-pgschemas#100] [Q&A] [answered]
I can get the subdomain approach to work but am having issues with the subfolder approach.

Using subdomain approach...
When I go to client1.localhost:8000 it takes me to the login screen for my app... http://client1.localhost:8000/accounts/login/?next=/
This works as expected.

Using subfolder approach...
When I go to clients.localhost:8000/client1/ it takes me to... http://clients.localhost:8000/accounts/login/?next=/client1/
This takes me to the "page not found error" and this message... No tenant for hostname 'clients.localhost'

How can I get the subfolder approach to work?

**Comment by lorinkoz** [comment, marked as answer]
Good day @dbill23. I transformed your issue into a Q/A discussion.

Could you please share your `TENANTS` settings, and an example of the `Domain` model instances you're creating?

**Reply by dbill23** [reply]
Thanks @lorinkoz for your reply. Some further points:

- I can add 'client1' into the http as follows, and this takes me to the logon screen: http://clients.localhost:8000/client1/accounts/login/?next=/client1/
- Once I'm logged in, I can navigate using the menu to other views and the folder approach seems to work fine ex: http://clients.localhost:8000/client1/tran/. But if I go to the logout or login option it seems to lose the tenant path (client1).

Here are my TENANT settings:
```
TENANTS = {
    "public": {
        "APPS": [
            "django.contrib.contenttypes",
            "django.contrib.staticfiles",
            "django_pgschemas",
            'customers',
        ],
    },
    "default": {
        "TENANT_MODEL": "customers.Client",
        "DOMAIN_MODEL": "customers.Domain",
        "APPS": [
            "django.contrib.auth",
            "django.contrib.sessions",
            "fin",
            'django_plotly_dash',
            'constance.backends.database', 
            'constance',
            'dal',
            'dal_select2',
            'django.contrib.admin',
            'django.contrib.contenttypes',
            'django.contrib.messages',
            'django.contrib.staticfiles',
            'fontawesomefree',
            'bootstrap_modal_forms',
            'widget_tweaks',
            'crispy_forms',
            'flatpickr',
            'bootstrap4',
            'django_filters',
            'django_extensions',
            'rest_framework',
        ],
        "URLCONF": "mysite.urls",
    }
}
```
Domain models:
domain: client1.localhost
folder:''
is_primary: false
redirect_to_primary: false
tenant_id: 5

domain: clients.localhost
folder: client1
is_primary: false
redirect_to_primary: false
tenant_id: 5

**Reply by lorinkoz** [reply]
Thanks for the info. There is a potential bug, as you mentioned, but I would need to reproduce locally to confirm. Stay tuned.

**Reply by lorinkoz** [reply]
Couple of additional things. Please make sure you:
1. Have included `django.contrib.auth.urls` in one of your urlconfs.
2. Have set `LOGIN_URL` in settings to `"login"` (because the [default value is hardcoded](https://docs.djangoproject.com/en/4.0/ref/settings/#login-url) to `"/accounts/login/"` )
